sycl/doc/SYCLPreprocessorMacroses.md

```diff
@@ -0,0 +1,22 @@
+# Overview
```

Comment:
Why is this file in this folder? Shouldn't this go into e.g. doc/extensions/contant_ptr?

Reply:
According to my understanding this not an extension of the functionality defined by the specification, but rather an implementation detail.

Reply:
To keep up with the spirit of "implementation details are DPC++", I'd suggest renaming this into `PreprocessorMacros.md`

Reply:
Sure.